src/Processors/Transforms/AggregatingInOrderTransform.cpp

```diff
@@ -82,23 +83,25 @@ void AggregatingInOrderTransform::consume(Chunk chunk)
         res_aggregate_columns.resize(params->params.aggregates_size);
 
         for (size_t i = 0; i < params->params.keys_size; ++i)
-        {
             res_key_columns[i] = res_header.safeGetByPosition(i).type->createColumn();
-        }
+
         for (size_t i = 0; i < params->params.aggregates_size; ++i)
-        {
             res_aggregate_columns[i] = res_header.safeGetByPosition(i + params->params.keys_size).type->createColumn();
-        }
+
         params->aggregator.createStatesAndFillKeyColumnsWithSingleKey(variants, key_columns, key_begin, res_key_columns);
+        params->aggregator.addArenasToAggregateColumns(variants, res_aggregate_columns);
         ++cur_block_size;
     }
+
     ssize_t mid = 0;
     ssize_t high = 0;
     ssize_t low = -1;
+
     /// Will split block into segments with the same key
     while (key_end != rows)
```

Comment:
Looks like this file was only refactored, no logic have changed (or I missed something?).
Maybe, we can extract binary search as a separate function. It could be more readable.

Reply:
The main change in this file is splitting `fillAggregateColumnsWithSingleKey` to 2 stages: `addArena` and `addKey`.

Reply:
Ok, that's logical.

Reply:
I've changed custom binary search to std::upper_bound.